challenge_serving_drinks/src/challenge_serving_drinks/sd_states.py

```diff
@@ -0,0 +1,483 @@
+        for i in range(0, head_samples):
+            if rospy.is_shutdown():
+                return
+
+            people_received = self.wait_for_cb()
+            rospy.logdebug('Got sample %d with seq %s', i, people_received.header.seq)
+            for person in people_received.people:
+                if {'RWave', 'LWave'}.intersection(set(person.tags)):
+                    waving_persons.append(person)
+
+            if waving_persons:
+                break
+        return people_received.header, waving_persons
+
+    def wait_for_cb(self):
+        timeout = 1
+
+        people_received = People()
+        while not rospy.is_shutdown() and not people_received.people:
+            try:
+                return self.people_queue.get(timeout=timeout)
+            except Empty:
+                rospy.logwarn('No people message received within %d seconds', timeout)
+
+
+class DescriptionStrDesignator(ds.Designator):
+    def __init__(self, message_type, drink_request_des, operator_name_des, name=None):
+        super(DescriptionStrDesignator, self).__init__(resolve_type=str, name=name)
+
+        ds.check_type(operator_name_des, str)
+        ds.check_type(drink_request_des, str)
+
+        self.message_type = message_type
+        self.operator_name_des = operator_name_des
+        self.drink_request_des = drink_request_des
+
+    def _resolve(self):
+        operator_name = self.operator_name_des.resolve()
+        drink_request = self.drink_request_des.resolve()
+        if self.message_type == "found_operator":
+            return "Hey {name}, I'm bringing your {drink}".format(name=operator_name, drink=drink_request)
+        elif self.message_type == "not_found_operator":
+            return "Hey {name} I cannot find you!"\
+                   "Please come to me to receive your {drink}".format(name=operator_name, drink=drink_request)
+        elif self.message_type == "fallback_bar":
+            return "Oh, I cannot inspect the bar. Please hand me over the {drink}".format(drink=drink_request)
+        elif self.message_type == "confirmation_available":
+            return "I understood that you would like {drink}, is that correct?".format(drink=drink_request)
+        elif self.message_type == "state_unavailable":
+            return random.choice([
+                    "Unfortunately we don't have {drink}".format(drink=drink_request),
+                    "The requested {drink} is unavailable".format(drink=drink_request),
+                    "I'm sorry but {drink} is out of stock".format(drink=drink_request),
+            ])
+        else:
+            rospy.logerr("No correct message type defined for DescriptionStrDesignator")
+
+
+if __name__ == "__main__":
+
+    # Test code
+
+    rospy.init_node('get_drinks')
+
+    from robot_skills.get_robot import get_robot_from_argv
+
+    _robot = get_robot_from_argv(index=1)
+
+    rospy.loginfo("Waiting for tf cache to be filled")
+    rospy.sleep(0.5)  # wait for tf cache to be filled
```

Comment:
Logging and comments is a bit double

Reply:
I'm not going to bother about this. Better to have too many comments than too few